pkg/util/hostname.go

```diff
@@ -30,8 +31,16 @@ var (
 		"localhost6.localdomain6",
 		"ip6-localhost",
 	}
+	hostnameExpvars  = expvar.NewMap("hostname")
+	hostnameProvider = expvar.String{}
+	hostnameError    = expvar.String{}
```

Comment:
We might encounter multiple errors, this should be a `map[string]string` so we could store every errors with their provider.

Reply:
The main goal is to explain why we choose a hostname instead of another one.

For example, when running on EC2 it the metadata endpoint from AWS is down we might fallback on the system hostname. In that case we want to display in the status page that the hostname provider was the system because EC2 returned an error.

Does that make sens ?